WordPress/Classes/ViewRelated/Stats/JetpackLoginViewController.swift

```diff
@@ -0,0 +1,546 @@
+        self.passwordTextField.placeholder = NSLocalizedString("WordPress.com password",
+                                                               comment: "Password placeholder")
+
+        self.verificationCodeTextField.delegate = self
+        self.verificationCodeTextField.font = WPNUXUtility.textFieldFont()
+        self.verificationCodeTextField.isHidden = true // Hidden by default
+        self.verificationCodeTextField.placeholder = NSLocalizedString("Verification Code",
+                                                                       comment: "Two factor code placeholder")
+        self.signinButton.isEnabled = false
+
+        setupSendSMSCodeButtonText()
+        self.sendSMSCodeButton.isHidden = true // Hidden by default
+
+        setupMoreInformationButtonText()
+        self.moreInformationButton.isHidden = true // Hidden by default
+
+        let title = NSLocalizedString("Install Jetpack", comment: "Title of a button for Jetpack Installation. The text " +
+                "should be uppercase.").localizedUppercase
+        self.installJetpackButton.setTitle(title, for: .normal)
+        self.installJetpackButton.isHidden = true // Hidden by default
+    }
+
+    /// Configures the button text that requests a 2fa code be sent via SMS.
+    ///
+    fileprivate func setupSendSMSCodeButtonText() {
+        let string = NSLocalizedString("Enter the code on your authenticator app or <u>send the code via text message</u>.",
+                                       comment: "Message displayed when a verification code is needed")
+
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.alignment = .center
+
+        let attributes: StyledHTMLAttributes = [ .BodyAttribute: [ NSFontAttributeName: UIFont.systemFont(ofSize: 14),
+                                                                   NSForegroundColorAttributeName: WPStyleGuide.allTAllShadeGrey(),
+                                                                   NSParagraphStyleAttributeName: paragraphStyle ]]
+
+        let attributedCode = NSAttributedString.attributedStringWithHTML(string, attributes: attributes)
+        let attributedCodeHighlighted = attributedCode.mutableCopy() as! NSMutableAttributedString
+        attributedCodeHighlighted.applyForegroundColor(WPNUXUtility.confirmationLabelColor())
+
+        if let titleLabel = self.sendSMSCodeButton.titleLabel {
+            titleLabel.lineBreakMode = .byWordWrapping
+            titleLabel.textAlignment = .center
+            titleLabel.numberOfLines = 3
+        }
+
+        self.sendSMSCodeButton.setAttributedTitle(attributedCode, for: UIControlState())
+        self.sendSMSCodeButton.setAttributedTitle(attributedCodeHighlighted, for: .highlighted)
+    }
+
+    /// Configures the button text for requesting more information about jetpack.
+    ///
+    fileprivate func setupMoreInformationButtonText() {
+        let string = NSLocalizedString("<u>More information</u>",
```

Comment:
The underline portion of this should be outside of the localized string IMHO. Also can we apply the underline with a font style instead of using HTML?